python/strings_udf/strings_udf/_typing.py

```diff
@@ -0,0 +1,354 @@
+# Copyright (c) 2022, NVIDIA CORPORATION.
+
+import operator
+
+import llvmlite.binding as ll
+from numba import types
+from numba.core.datamodel import default_manager
+from numba.core.extending import models, register_model
+from numba.core.typing import signature as nb_signature
+from numba.core.typing.templates import AbstractTemplate, AttributeTemplate
+from numba.cuda.cudadecl import registry as cuda_decl_registry
+from numba.cuda.cudadrv import nvvm
+
+data_layout = nvvm.data_layout
+
+# workaround for numba < 0.56
+if isinstance(data_layout, dict):
+    data_layout = data_layout[64]
+target_data = ll.create_target_data(data_layout)
+
+
+# String object definitions
+class DString(types.Type):
+    def __init__(self):
+        super().__init__(name="dstring")
+        llty = default_manager[self].get_value_type()
+        self.size_bytes = llty.get_abi_size(target_data)
+
+
+class StringView(types.Type):
+    def __init__(self):
+        super().__init__(name="string_view")
+        llty = default_manager[self].get_value_type()
+        self.size_bytes = llty.get_abi_size(target_data)
+
+
+@register_model(StringView)
+class stringview_model(models.StructModel):
+    # from string_view.hpp:
+    _members = (
+        # const char* _data{}
+        # Pointer to device memory contain char array for this string
+        ("data", types.CPointer(types.char)),
+        #  size_type _bytes{};
+        #  Number of bytes in _data for this string
+        ("bytes", types.int32),
+        #  mutable size_type _length{};
+        #  Number of characters in this string (computed)
```

Comment:
Seems like there's an extra space on these lines.
```suggestion
        # size_type _bytes{};
        # Number of bytes in _data for this string
        ("bytes", types.int32),
        # mutable size_type _length{};
        # Number of characters in this string (computed)
```